javascript/tables/src/validateTable.ts

```diff
@@ -0,0 +1,288 @@
+						? (columnValues as Date[])
+						: [],
+				)
+				if (!minimumResult) {
+					flag = true
+
+					if (validationResult.columnErrors != null)
+						validationResult.columnErrors.push(
+							createColumnErrorObject(field.name, 'minimum'),
+						)
+				}
+			}
+
+			//maximum constraint - for numbers or dates only
+			if (
+				!flag &&
+				constraints.maximum != null &&
+				constraints.maximum &&
+				(field.type === DataType.Number || field.type === DataType.Date)
+			) {
+				const maximumResult = checkMaximumConstraint(
+					constraints.maximum,
+					field.type === DataType.Number
+						? (columnValues as number[])
+						: field.type === DataType.Date
+						? (columnValues as Date[])
+						: [],
+				)
+				if (!maximumResult) {
+					flag = true
+
+					if (validationResult.columnErrors != null)
+						validationResult.columnErrors.push(
+							createColumnErrorObject(field.name, 'maximum'),
+						)
+				}
+			}
+
+			//pattern constraint - for strings only
+			if (
+				!flag &&
+				constraints.pattern != null &&
+				constraints.pattern &&
+				field.type === DataType.String
+			) {
+				const patternResult = checkPatternConstraint(
+					constraints.pattern,
+					columnValues as string[],
+				)
+				if (!patternResult) {
+					flag = true
+
+					if (validationResult.columnErrors != null)
+						validationResult.columnErrors.push(
+							createColumnErrorObject(field.name, 'pattern'),
+						)
+				}
+			}
+
+			//enum constraint - for strings only
+			if (
+				!flag &&
+				constraints.enum != null &&
+				constraints.enum &&
+				field.type === DataType.String
+			) {
+				const enumResult = checkEnumConstraint(
+					constraints.enum,
+					columnValues as string[],
+				)
+				if (!enumResult) {
+					flag = true
+
+					if (validationResult.columnErrors != null)
+						validationResult.columnErrors.push(
+							createColumnErrorObject(field.name, 'enum'),
+						)
+				}
+			}
+		}
+	})
+
+	return validationResult
+}
+
+function createColumnErrorObject(name: string, message: string): ColumnError {
+	const columnError: ColumnError = {
+		columnName: name,
+		rule: message,
+	}
+
+	return columnError
+}
+
+function checkRequiredConstraint(values: string[]): boolean {
+	for (let i = 0; i < values.length; i++) {
+		if (
+			values[i] == null ||
+			values[i] == undefined ||
+			(values[i] as string).length == 0
+		)
+			return false
+	}
+
+	return true
+}
+
+function checkUniqueConstraint(values: string[]): boolean {
+	const uniqueValues = [...new Set(values)]
+	return values.length === uniqueValues.length
+}
+
+function checkMinLengthConstraint(
+	minLength: number,
+	values: string[] | Array<any>[],
+): boolean {
+	for (let i = 0; i < values.length; i++) {
+		if (values[i] != null && values[i]!.length < minLength) return false
+	}
+
+	return true
+}
+
+function checkMaxLengthConstraint(
+	maxLength: number,
+	values: string[] | Array<any>[],
+): boolean {
+	for (let i = 0; i < values.length; i++) {
```

Comment:
`values.every`

Reply:
And the rest, I see there are several more loops so I'll stop adding comments :)